src/Nethermind/Nethermind.Merge.AuRa.Test/AuRaMergeEngineModuleTests.cs

```diff
@@ -87,14 +99,138 @@ public override Task Can_apply_withdrawals_correctly((Withdrawal[][] Withdrawals
+    {
+        using MergeTestBlockchain chain = await new MergeAuRaTestBlockchain(null, null, true).Build(new TestSingleReleaseSpecProvider(London.Instance));
+        IEngineRpcModule rpc = CreateEngineModule(chain);
+
+        // creating chain with 3 blocks
+        IReadOnlyList<ExecutionPayload> executionPayloads = await ProduceBranchV1(rpc, chain, 3, CreateParentBlockRequestOnHead(chain.BlockTree), true);
+        _parentPayload = executionPayloads.Last();
+
+        // should contain shutter transactions
+        executionPayloads[0].GetTransactions().Should().HaveCount(1);
+        executionPayloads[0].GetTransactions()[0].Value.Should().Be(123);
+
+        executionPayloads[1].GetTransactions().Should().HaveCount(1);
+        executionPayloads[1].GetTransactions()[0].Value.Should().Be(124);
+
+        executionPayloads[2].GetTransactions().Should().HaveCount(1);
+        executionPayloads[2].GetTransactions()[0].Value.Should().Be(125);
+
+        // build blocks with transactions from both shutter and TxPool
+        // id is used as value to differentiate between transactions
+        (int id0, Transaction[] transactions0) = await ProduceBlockWithTransactions(rpc, chain, 3);
+        transactions0.Should().HaveCount(4);
+        transactions0[0].Value.Should().Be(126);
+        transactions0[1].Value.Should().Be(id0.GWei());
+        transactions0[2].Value.Should().Be(id0.GWei());
+        transactions0[3].Value.Should().Be(id0.GWei());
+
+        (int id1, Transaction[] transactions1) = await ProduceBlockWithTransactions(rpc, chain, 3);
+        transactions1.Should().HaveCount(4);
+        transactions1[0].Value.Should().Be(127);
+        transactions1[1].Value.Should().Be(id1.GWei());
+        transactions1[2].Value.Should().Be(id1.GWei());
+        transactions1[3].Value.Should().Be(id1.GWei());
+
+        (int id2, Transaction[] transactions2) = await ProduceBlockWithTransactions(rpc, chain, 1);
+        transactions2.Should().HaveCount(2);
+        transactions2[0].Value.Should().Be(128);
+        transactions2[1].Value.Should().Be(id2.GWei());
+
+        (int id3, Transaction[] transactions3) = await ProduceBlockWithTransactions(rpc, chain, 0);
+        transactions3.Should().HaveCount(1);
+        transactions3[0].Value.Should().Be(129);
+
+        (int id4, Transaction[] transactions4) = await ProduceBlockWithTransactions(rpc, chain, 2);
+        // bad shutter transaction excluded
+        transactions4[0].Value.Should().Be(id4.GWei());
+        transactions4[1].Value.Should().Be(id4.GWei());
+
+        (int id5, Transaction[] transactions5) = await ProduceBlockWithTransactions(rpc, chain, 4);
+        // bad shutter transaction excluded
+        transactions5.Should().HaveCount(4);
+        transactions5[0].Value.Should().Be(id5.GWei());
+        transactions5[1].Value.Should().Be(id5.GWei());
+        transactions5[2].Value.Should().Be(id5.GWei());
+        transactions5[3].Value.Should().Be(id5.GWei());
```

Comment:
Shouldn't those be multiple tests?
We could use collections assertions instead of doing multiple assertions per collection